src/rpc/server.cpp

```diff
@@ -546,6 +556,25 @@ UniValue CRPCTable::execute(const JSONRPCRequest &request) const
     if (!pcmd)
         throw JSONRPCError(RPC_METHOD_NOT_FOUND, "Method not found");
 
+    // Before executing the RPC Command, filter commands from platform rpc user
+    if (fMasternodeMode && request.authUser == gArgs.GetArg("-platform-user", defaultPlatformUser)) {
+        auto it = find_if(platformAllowedCommands.begin(), platformAllowedCommands.end(), [request](const std::pair<std::string /*command*/, std::string /*subcommand*/>& cmd) {
+            // Check if this request matches a valid masternode rpc command
+            if (request.strMethod == cmd.first) {
+                // First arg / command matched, if there is no subcommand return true
+                if (cmd.second.empty()) return true;
+                if (request.params[0].isStr() && request.params[0].getValStr() == cmd.second) {
+                    // Sub command was needed and matched, return true
+                    return true;
+                }
+            }
+            return false;
```

Comment:
Too many if-s imo :) maybe smth like this
```suggestion
            // Check if this request matches a valid platform allowed rpc command
            return (request.strMethod == cmd.first) &&
                // and there is no subcommand
                (cmd.second.empty() ||
                // or a subcommand was needed and it matches too
                (request.params[0].isStr() && request.params[0].getValStr() == cmd.second));
```
instead?

Reply:
Hmm, idk, I find the logic flow of the if statements easier to understand

Reply:
Nah, nevermind, I think this change makes sense :+1: